site/content/en/latest/user/private-key-provider.md

```diff
@@ -0,0 +1,368 @@
+---
+title: "Accelerating TLS Handshakes using Private Key Provider in Envoy"
+---
+
+TLS operations can be accelerated or the private key can be protected using specilized hardware. This can be leveraged in Envoy using [Envoy Private Key Provider](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#extensions-transport-sockets-tls-v3-privatekeyprovider) is added to Envoy.
+
+Today, there are two private key providers implemented in Envoy as contrib extensions:
+- [QAT in Envoy 1.24 release](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/private_key_providers/qat/v3alpha/qat.proto#extensions-private-key-providers-qat-v3alpha-qatprivatekeymethodconfig)
+- [CryptoMB in Envoy 1.20 release](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/private_key_providers/cryptomb/v3alpha/cryptomb.proto )
+
+Both of them are used to accelerate the TLS handshake through the hardware capabilities.
+
+This guide will walk you through the steps required to configure TLS Termination mode for TCP traffic while also using the Envoy Private Key Provider to accelerate the TLS handshake.
+
+## Prerequisites
+
+### For QAT
+
+- Install Linux kernel 5.17 or similar
+- Ensure the node has QAT devices by checking the QAT physical function devices presented.
+
+  ```shell
+  lspci -d 8086:4940
+  ```
+
+- Enable IOMMU from BIOS
+- Enable IOMMU for Linux kernel
+
+  ```shell
+  cat /etc/default/grub:
+  GRUB_CMDLINE_LINUX="intel_iommu=on vfio-pci.ids=8086:4941"
+  update-grub
+  reboot
+  ````
+
+  Once the system is rebooted, check if the IOMMU has been enabled via the following command:
+
+  ```shell
+  dmesg| grep IOMMU
+  [    1.528237] DMAR: IOMMU enabled
+  ```
+
+- Enable virtual function devices for QAT device
+
+  ```shell
+  rmmod qat_4xxx
+  modprobe qat_4xxx
+  qat_device=$(lspci -D -d :4940 | awk '{print $1}')
+  for i in $qat_device; do echo 16|sudo tee /sys/bus/pci/devices/$i/sriov_numvfs; done
+  ```
```

Comment:
Missing `modprobe vfio_pci` at the beginning and `chmod a+rw /dev/vfio/*` at the end?

Reply:
thanks！